runtime/tr.source/trj9/optimizer/SPMDParallelizer.cpp

```diff
@@ -2641,48 +2641,33 @@ void TR_SPMDKernelParallelizer::insertGPUTemporariesLivenessCode(List<TR::TreeTo
 
    for (TR::TreeTop *insertionPoint = lit.getFirst(); insertionPoint; insertionPoint = lit.getNext())
       {
-      int i = 0;
-      TR::Node* currentNode;
-      TR::Node* prevNode;
+      TR::Node* currentNode = 0;
 
-      if (!firstKernel)
+      if (firstKernel)
+         {
+         currentNode = TR::Node::create(insertionPoint->getNode(), TR::lconst, 0, 0);
```

Comment:
This _may_ not be correct. You may have to use `setLongInt`. However we already have a nice API defined for creating `lconst` nodes here:
https://github.com/eclipse/omr/blob/381ec5b6fb6454f94d78a12b4349c32095e016a2/compiler/il/OMRNode.cpp#L1369

We should try to use this API to avoid these sorts of issues.

Reply:
That sounds like a good idea to me. Is the previous version actually an error? I see other places in the codebase that do a similar thing where it does not call setLongInt.

Reply:
@IBMJimmyk yes this is a bug. If you look at the overload which gets called, it is this one:
https://github.com/eclipse/omr/blob/381ec5b6fb6454f94d78a12b4349c32095e016a2/compiler/il/OMRNode.cpp#L621

The overload takes an `int32_t` so if the `lconst` we're trying to initialize exceeds the 32-bit range there will be problems. Can you open up an issue on OMR and OpenJ9 for this to be addressed in the places you saw?

Reply:
My updated change now uses lconst(). I will doublecheck my findings and then create a new issue regarding other locations in the code that may not be creating lconst nodes correctly.